src/components/MyAccount/CheckoutsTab.tsx

```diff
@@ -0,0 +1,243 @@
+        >
+          <Icon
+            size="large"
+            name="actionCheckCircleFilled"
+            color="ui.success.primary"
+            sx={{ marginTop: "4px" }}
+          />
+          <Text sx={{ marginBottom: "0px" }}> Renewal successful </Text>
+        </Box>
+      ),
+    }
+    const failureModalProps = {
+      bodyContent: (
+        <Box
+          sx={{
+            fontSize: "desktop.body.body1.light",
+            display: "flex",
+            marginTop: "-xs",
+          }}
+        >
+          <Text sx={{ marginLeft: "l", marginRight: "m" }}>
+            We were unable to renew your item. Please try again or{" "}
+            <Link href="https://www.nypl.org/get-help/contact-us">
+              contact us
+            </Link>{" "}
+            for assistance.
+          </Text>
+        </Box>
+      ),
+      closeButtonLabel: "OK",
+      headingText: (
+        <Box
+          sx={{
+            fontSize: "desktop.heading.heading5",
+            paddingTop: "m",
+            display: "flex",
+            justifyContent: "flex-start",
+            gap: "10px",
+          }}
+        >
+          <Icon
+            size="large"
+            name="errorFilled"
+            color="ui.error.primary"
+            sx={{ marginTop: "4px" }}
+          />
+          <Text sx={{ marginBottom: "0px" }}> Renewal failed </Text>
+        </Box>
+      ),
+    }
+
+    useEffect(() => {
+      const lastDisabledTime = localStorage.getItem(
+        `lastDisabledTime-${checkout.id}`
+      )
+      if (lastDisabledTime) {
+        const twentyFourHoursAgo = new Date().getTime() - 8640000
+        if (parseInt(lastDisabledTime, 10) > twentyFourHoursAgo) {
+          setButtonDisabled(true)
+        }
+      }
+    }, [])
+
+    const handleClick = async () => {
+      const response = await fetch(
+        `${BASE_URL}/api/account/checkouts/renew/${checkout.id}`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ patronId: patron.id }),
+        }
+      )
+      const responseData = await response.json()
+      if (responseData.message == "Renewed") {
+        setButtonDisabled(true)
+        setModalProps(successModalProps)
+        onOpen()
+        localStorage.setItem(
+          `lastDisabledTime-${checkout.id}`,
+          new Date().getTime().toString()
+        )
+      } else {
+        setModalProps(failureModalProps)
+        onOpen()
+        //TO-DO: Log error console.log("error", responseData)
+      }
+    }
+
+    return (
+      <>
+        <Button
+          sx={{
+            width: "-webkit-fill-available",
+          }}
+          buttonType="secondary"
+          id={`renew-${checkout.id}`}
+          onClick={handleClick}
+          isDisabled={isButtonDisabled}
+        >
+          Renew
+        </Button>
+        <Modal {...modalProps} />
+      </>
+    )
+  }
+
+  function getDueDate(date) {
```

Comment:
can `getDueDate`, `getTitle` live on the Patron class so the data is passed to this component without needing a transformation?